e2e-tests/docs/CI.md

```diff
@@ -0,0 +1,135 @@
+
+### Retrying Tests
+
+PR tests are not automatically retried beyond the individual test retries specified in the Playwright configuration (each test is retried up to 2 times on failure). However, you can manually retrigger the entire test suite by commenting:
+
+- `/retest e2e-tests`
+- `/test all`
+
+This is useful if you believe a failure was due to a flake or external issue and want to rerun the tests without making any code changes.
+
+### CI Job Definitions
+
+#### Pull Request Test Job
+
+- **Purpose:** Validate new PRs for code quality, functionality, and integration.
+- **Trigger:** When a PR is opened and `/ok-to-test` is commented by a janus-idp member, or when `/test`, `/test all`, or `/retest` is issued after validation.
+- **Environment:** Runs on the ephemeral `rhdh-pr-os` cluster on IBM Cloud.
+- **Configurations:**
+  - Tests are executed on both **RBAC** (Role-Based Access Control) and **non-RBAC** namespaces.
+- **Steps:**
+  1. **Detection:** OpenShift-CI detects the PR event.
+  2. **Environment Setup:** The test environment is set up using the `openshift-ci-tests.sh` script.
+  3. **Test Execution:**
+     - **Running Tests:** Executes test suites using `yarn` commands specified in `package.json`.
+     - **Retry Logic:** Individual tests are retried up to 2 times as specified in the Playwright configuration.
+  4. **Artifact Collection:**
+     - Collects test artifacts (logs, screenshots, recordings).
+     - Stores artifacts in the designated `ARTIFACT_DIR` for a retention period of **6 months**.
+  5. **Reporting:**
+     - Reports status back to the PR checks.
+     - Generates and uploads HTML reports.
+- **Artifacts:** Test reports, logs, screenshots, accessible via PR details under **Artifacts**.
+- **Notifications:** Status updates posted on the PR.
+- **Manual Retriggering:**
+  - Tests can be manually retriggered using the `/retest e2e-tests` or `/test all` commands in the PR comments.
+
+### GitHub PRs Diagram
+
+![GitHub PR Testing Diagram](./resources/github_diagram.svg)
+
+## Nightly Tests
+
+Nightly tests are run to ensure the stability and reliability of our codebase over time. These tests are executed on different clusters to cover various environments, including both **RBAC** and **non-RBAC**.
+
+- **AKS Nightly Tests:** Nightly tests for Azure Kubernetes Service (AKS) run on the `bsCluster`. We do not have AKS PR checks; the AKS environment is exclusively used for nightly runs.
+
+- **IBM Cloud Tests:** All nightly tests for the `main`, `1.3`, and `1.2` branches run against the `rhdh-pr-os` OpenShift Container Platform (OCP) cluster on IBM Cloud.
```

Comment:
Correct branch names are 'main', 'release-1.3' and '1.2.x'

Reply:
fixed!